daemon/graphdriver/counter.go

```diff
@@ -24,29 +24,19 @@ func NewRefCounter(c Checker) *RefCounter {
 
 // Increment increaes the ref count for the given id and returns the current count
 func (c *RefCounter) Increment(path string) int {
-	c.mu.Lock()
-	m := c.counts[path]
-	if m == nil {
-		m = &minfo{}
-		c.counts[path] = m
-	}
-	// if we are checking this path for the first time check to make sure
-	// if it was already mounted on the system and make sure we have a correct ref
-	// count if it is mounted as it is in use.
-	if !m.check {
-		m.check = true
-		if c.checker.IsMounted(path) {
-			m.count++
-		}
-	}
-	m.count++
-	count := m.count
-	c.mu.Unlock()
-	return count
+	return c.operation(path, func(minfo *minfo) {
+		minfo.count++
+	})
 }
 
 // Decrement decreases the ref count for the given id and returns the current count
 func (c *RefCounter) Decrement(path string) int {
+	return c.operation(path, func(minfo *minfo) {
+		minfo.count--
+	})
+}
+
+func (c *RefCounter) operation(path string, infoOp func(minfo *minfo)) int {
```

Comment:
I wonder it is too much to have general func here.
I think `func (c *RefCounter) incdec(path string, inc bool)) int` is ok